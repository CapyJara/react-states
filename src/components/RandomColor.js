import React, { PureComponent } from 'react';
const chance = require('chance').Chance();

export default class RandomColor extends PureComponent {
  state = {
    color: '',
    trigger: new Date(),
    img: ''
  }

  randomColor = () => {
    const color = chance.pickone(['red', 'blue', 'green', 'yellow', 'orange', 'purple']);
    return color;
  }

  componentDidMount = () => {
    setInterval(() => {
      this.setState({ color: this.randomColor(), trigger: new Date(), img: '' });
    }, 1000);
  }

  componentDidUpdate(prevProps, prevState) {
    if(prevState.color === this.state.color) {
      this.setState({ img: 'https://thumbs.gfycat.com/DemandingWelltodoGallinule-poster.jpg' });
    }
  }

  render() {
    const { color, img } = this.state;
    const styles = {
      background: img ? `center / contain url(${img}` : color,
      width: '50em',
      height: '50em'
    };
    return <div style={styles}></div>;
  }
}
