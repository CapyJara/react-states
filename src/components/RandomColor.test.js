import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('App component', () => {
  it('renders a App', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
