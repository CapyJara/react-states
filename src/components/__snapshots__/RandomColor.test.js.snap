// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders a App 1`] = `
<div
  style={
    Object {
      "background": "",
      "height": "50em",
      "width": "50em",
    }
  }
/>
`;
